app/assets/javascripts/auth_source_ldap.js

```diff
@@ -15,4 +15,33 @@ function change_ldap_port(item) {
       }
     }
   }
+}
+function test_connection(test_connection_url){
+    if($("#aid_test_connection_button_auth_source_ldap").prop('disabled') != true ){
+      var serialized_form_data = $("form").serializeObject();
+      serialized_form_data["_method"]="post";
+
+      $("#aid_test_connection_button_auth_source_ldap").prop('disabled',true);
+
+      $.ajax({
+        url : test_connection_url,
+        type : 'PUT',
+        data : serialized_form_data,
+        success :  function(data){
+            $("#aid_test_connection_button_auth_source_ldap").prop('disabled',false);
+            if(data.success){
+              notify("<div>"+data.message+"</div>","success");
+            } else{
+              // In the case of error in connection to the ldap server, we are using the $.jnotify function directly instead of the wrapper "notify(item, type)" 
+              // because we will need some HTML formatting for the error to be more readable
+              var error_message = "<strong>"+__("Error in establishing connection :: ")+"</strong>";
+              error_message += "<br/><br/>";
+              error_message += "<i>"+ data.error_class + "</i>"
+              error_message += " => "
+              error_message += "<u>" + data.message + "</u>";
```

Comment:
Maybe remove some of the italics and underlining, keep it simpler for consistency?  e.g. a single `<br> + data.error_class + ": " + data.message`